CalibTracker/SiStripHitEfficiency/plugins/HitEff.cc

```diff
@@ -369,15 +377,55 @@ void HitEff::analyze(const edm::Event& e, const edm::EventSetup& es) {
 
       // Check whether the trajectory has some missing hits
       bool hasMissingHits = false;
-      for (itm = TMeas.begin(); itm != TMeas.end(); itm++) {
-        auto theHit = (*itm).recHit();
+      unsigned int previous_layer = 999;
+      vector<unsigned int> missedLayers;
+
+      for (const auto& itm : TMeas) {
+        auto theHit = itm.recHit();
+        unsigned int iidd = theHit->geographicalId().rawId();
+        unsigned int layer = ::checkLayer(iidd, tTopo);
+        int missedLayer = int(layer + 1);
+        int diffPreviousLayer = int(layer - previous_layer);
+        if (doMissingHitsRecovery_) {
+          //Layers from TIB + TOB
+          if (diffPreviousLayer == -2 && missedLayer > k_LayersStart && missedLayer < k_LayersAtTOBEnd) {
+            missHitPerLayer[missedLayer] += 1;
+            hasMissingHits = true;
+          }
+          //Layers from TID
+          if (diffPreviousLayer == -2 && (missedLayer == 12 || missedLayer == 13)) {
```

Comment:
Not obvious whether `missedLayer == 12` or `missedLayer == 13` can enter in the previous condition.
Perhaps an `else if` (here and at [L401](https://github.com/cms-sw/cmssw/pull/41970/files#diff-2605c956edf7b5d5b9ae0863fc9f246940b9ba7274eb40a0747fcd088fe7a3c2R401)) can help clarifying

Reply:
```suggestion
          else if (diffPreviousLayer == -2 && missedLayer > k_LayersAtTOBEnd +1 && missedLayer <= k_LayersAtTIDEnd) {
```
(by the way, layer = 11 is not in TID?)

Reply:
Yes, but with the condition "diffPreviousLayer == -2", the layer = 11 will never enter in this case